security/cmd/node_agent/na/nodeagent.go

```diff
@@ -123,7 +123,7 @@ func (na *nodeAgentInternal) createRequest() ([]byte, *pb.CsrRequest, error) {
 		RSAKeySize: na.config.RSAKeySize,
 	})
 	if err != nil {
-		return nil, nil, fmt.Errorf("request creation fails on CSR generation (%v)", err)
```

Comment:
I would suggest to keep this line for more debugging info.

Reply:
Yes. I originally thought the same, but it turned out this wrapping made the error message extra verbose, the error message looks like:
"request creation fails on CSR generation (CSR generation fails at X509 cert request generation (crypto/rsa: message too long for RSA public key size))"
While without this wrapping, the error message looks like:
"CSR generation fails at X509 cert request generation (crypto/rsa: message too long for RSA public key size)"
I personally think it's better to remove the wrapping here :)